LinkedList/LinkedList/TSLList.h

```diff
@@ -0,0 +1,350 @@
+	{
+		FNode* Temporal = OtherTSLlist.Head;
+
+		for (int i = 0; i < Size; i++)
+		{
+			AddTail(Temporal->Element);
+			Temporal = Temporal->Next;
+		}
+	}
+
+	//--------------------------------FIterator
+	class FIterator
+	{
+	private:
+		FNode* Current;
+
+	public:
+		FIterator(FNode* NewCurrent)
+		{
+			Current = NewCurrent;
+		}
+
+		FIterator& operator++()
+		{
+			Current = Current->Next;
+			return *this;
+		}
+
+		Type& operator*()
+		{
+			return Current->Element;
+		}
+
+		const Type& operator*() const
+		{
+			return Current->Element;
+		}
+
+		bool operator!=(const FIterator& Other)
+		{
+			return Current != Other.Current;
+		}
+
+		bool operator!=(const FIterator& Other) const
+		{
+			return Current != Other.Current;
+		}
+	};
+
+	FIterator begin()
+	{
+		return FIterator(Head);
+	}
+
+	FIterator end()
+	{
+		return FIterator(nullptr);
+	}
+
+	//-------------------------Access
+	Type& operator[] (const int Index)
+	{
+		FNode* Temporal = Head; 
+		for (int i = 0; i < Index; i++)
+		{
+			Temporal = Temporal->Next;
+		}
+		return Temporal->Element;
+	}
+
+	const Type& operator[] (const int Index) const
+	{
+		FNode* Temporal = Head;
+		for (int i = 0; i < Index; i++)
+		{
+			Temporal = Temporal->Next;
+		}
+		return Temporal->Element;
+	}
+
+	//-----------------------------------------------
+	TSLList& operator= (const TSLList<Type>& OtherTSLlist)
+	{
+		Clear();
+		FNode* CurrentNode = OtherTSLlist.Head;
+
+		for (int i = 0; i < Size; i++)
+		{
+			AddTail(CurrentNode->Element);
+			CurrentNode = CurrentNode->Next;
+		}
+		return *this;
+
+	}
+
+	//-----------------------------------------------
+	Type& GetHead()
+	{
+		return Head->Element;
+	}
+
+	const Type& GetHead() const
+	{
+		return Head->Element;
+	}
+
+	//-----------------------------------------------
+	Type& GetTail()
+	{
+		return Tail->Element;
+	}
+
+	const Type& GetTail() const
+	{
+		return Tail->Element;
+	}
+
+	//---------------------------------Size
+	bool IsEmpty()
+	{
+		return Size == 0;
+	}
+
+	//------------------------------------------------
+	int GetSize()
+	{
+		return Size;
+	}
+
+	//-----------------------------------Modifiers
+	void AddHead(const Type& InHead)
+	{
+		FNode* NewHead = new FNode;
+		NewHead->Element = InHead;
+
+		if (Head == nullptr)
+		{
+			Head = NewHead;
+			Tail = NewHead;
+			Size++;
+		}
+		else
+		{
+			NewHead->Next = Head;
+			Head = NewHead;
+			Size++;
+		}
+	}
+
+	//-------------------------------------------------
+	void AddTail(const Type& InTail)
+	{
+
+		if (Head == nullptr)
+		{
+			AddHead(InTail);
+		}
+		else
+		{
+			Tail->Next = new FNode;
+			Tail->Next->Element = InTail;
+			Tail = Tail->Next;
+			Size++;
+		}
+	}
+	
+	//----------------------------------------------------
+	void Insert(const Type& NewItem, const int Index)
+	{
+		if (Index == 0)
+		{
+			AddHead(NewItem);
+		}
+		else if(Index == Size)
+		{
+			AddTail(NewItem);
+		}
+
+		else if (Index > 0 && Index < Size)
```

Comment:
I think to check on bounds first might be best, that way you could simply exit the function on such cases, else you will need to check 3 conditions before exiting